app/controllers/health_check_controller.rb

```diff
@@ -0,0 +1,100 @@
+class HealthCheckController < HealthCheck::HealthCheckController
+  # This overwrites the original HealthCheckController from the health_check gem, in order to add
+  # a custom check agaist the utilty's API health check. Original controller:
+  # https://github.com/ianheggie/health_check/blob/rails5/lib/health_check/health_check_controller.rb
+  rescue_from ActiveRecord::RecordNotFound, with: :render_nothing_not_found
+
+  def index
+    return super if utility.blank?
+    return render_failed_utility_inactive unless utility.active
+    return super if utility.health_check_url.blank?
+
+    @utility_response = utility.utility_service.health_check
+    return super if @utility_response.code == 200
+
+    render_failed_utility_health_check
+  end
+
+  private
+
+  def utility
+    utility_code_header = request.headers['Utility-ID']
+    return if utility_code_header.nil?
+
+    utility_code = sanitized_utility_code(utility_code_header)
+    @utility ||= Utility.find_by!(code: utility_code)
+  end
+
+  def render_failed_utility_inactive
+    @utility_error = false
+    @utility_disabled = true
+    msg = "health_check failed: #{utility.name} API is disabled."
+    # Log a single line as some uptime checkers only record that it failed, not the text returned
+    logger&.info msg
+    send_response(
+      msg, HealthCheck.http_status_for_error_text, HealthCheck.http_status_for_error_object,
+      display_message: utility.display_downtime_message
+    )
+  end
+
+  def render_failed_utility_health_check
+    @utility_error = true
+    @utility_disabled = false
+    msg = "health_check failed: #{utility.name} API is down."
+    # Log a single line as some uptime checkers only record that it failed, not the text returned
+    logger&.info msg
+    send_response(
+      msg, HealthCheck.http_status_for_error_text, HealthCheck.http_status_for_error_object,
+      display_message: utility.display_downtime_message
+    )
+  end
+
+  def sanitized_utility_code(utility_code_header)
+    Integer(utility_code_header)
+  rescue ArgumentError
+    nil
+  end
+
+  def render_nothing_not_found
+    head :not_found
+  end
+
+  def update_utility_active(value)
+    utility.update(active: value)
+  end
+
+  def send_response(msg, text_status, obj_status, display_message: nil)
+    obj = build_obj(msg, display_message)
+    msg ||= HealthCheck.success
+    respond_to do |format|
+      format.html { render plain_key => msg, status: text_status, content_type: 'text/plain' }
+      format.json { render json: obj, status: obj_status }
+      format.xml { render xml: obj, status: obj_status }
+      format.any { render plain_key => msg, status: text_status, content_type: 'text/plain' }
```

Comment:
A little doubt:
Do we need all these formats? Or only json?

Reply:
I think that we want always json, but I'm not sure about why are using other formats.
I will check that! 🤗

Reply:
We talked with Nacho abaout this. The default json is setting on routes.rb, but it's okey to keep all formats possibilities 🤗